### ISSUE TITLE
Use docker image for stable Windows packaging jobs on GitLab CI

### ISSUE THREAD

**Issue by Zimmi48** [coq/coq#7428]
Once #7344 gets in, GitLab CI will be used to build Windows packages (in place of AppVeyor).

There is a long-standing idea that we could make these builds more reliable by using a fixed set of dependencies that we would store somewhere. (This would also mean that the open source archive doesn't have to be regenerated at each build.)

Following #7387, I suggested trying to use Docker on Windows as well for this purpose. This would mean that GitLab CI would pull a docker image containing all the required dependencies for the build from, e.g., `dockerhub.com/coqci/coq`. (Maybe we can even get the Windows runner to cache this image.) This image would only be rebuilt once in a while (likely after each release) using a recipe that would be stored in Coq `dev/ci`.

@MSoegtropIMC has some experience with Docker, so he agreed to look into this after #7344 is merged.

**Comment by ejgallego** [comment]
That sounds good and even good practice I'd say. Let me know your docker handles so I add you to the `coqci` organization by the way.

**Comment by Zimmi48** [comment]
Because our packaging infrastructure is using our own VMs on which we install `git`, `7zip` and GitLab-specific stuff, we could also install all the required dependencies of Coq (only once), generate the opensource-archive (only once), then have the CI job only compile Coq itself and upload the corresponding artifact. This would basically mean getting all the benefits above (except the more principled approach which comes with Docker). So if using Docker is too difficult, this can be considered as a possible alternative.

**Comment by Zimmi48** [comment]
That was meant as a second option if the first didn't work out, and I only wrote it down here so that we wouldn't forget.

**Comment by ejgallego** [comment]
I suggest we close this in favor of caching the cygwin and opam install of the platform build, which should be doable, once #12425 is merged.

**Comment by Zimmi48** [comment]
You've marked this as closed by #12425. Didn't you intend to keep it open to track the caching aspect?

**Comment by ejgallego** [comment]
>  Didn't you intend to keep it open to track the caching aspect?

We won't be using docker for the caching so I think that deserves a new issue, and this one is closed due to removal from Gitlab CI.
